nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RPathUtils.java

```diff
@@ -75,17 +82,30 @@ static String packagesGzPath(final String path) {
    * Extracts full path from {@link Context}
    */
   static String extractFullPath(final Context context) {
-    return removeInitialSlashFromPath(context.getRequest().getPath());
+    return StringUtils.stripStart(context.getRequest().getPath(), "/");
   }
 
   /**
-   * Removes slash if path starts with it
+   * Extracts full path from {@link Context}
    */
-  private static String removeInitialSlashFromPath(final String path) {
-    if (path.startsWith("/")) {
-      return path.replaceFirst("/", "");
+  static String cutFilenameFromPath(final String path) {
+    final int pathEndCharIndex = StringUtils.stripEnd(path, "/").lastIndexOf('/');
+    return path.substring(0, pathEndCharIndex);
+  }
+
+  /**
+   * Determines asset kind by it's path
+   */
+  static AssetKind getAssetKind(final String path) {
+    AssetKind assetKind = ARCHIVE;
```

Comment:
Question: Is it safe to assume `ARCHIVE`? is it possible for an invalid path to be here?

Reply:
Yes, cause this method looks only on stored assets. So if it's not any kind of metadata it's definitely archive.

Reply:
BTW, I just moved this method without big modification